### ISSUE TITLE
Parsing MGRS strings with "ME" or "MN" square identification gives wrong results

### ISSUE THREAD

**Issue by B4rrCH** [Tronald/CoordinateSharp#171]
**Describe the bug**
MGRS coordinate strings with "ME" or "MN" 100,000-meter square identification fail to be parsed under some circumstances (namely when formatted like Coordinate.MGRS.ToString()).

**To Reproduce**
```csharp
Coordinate c1 = Coordinate.Parse("23J MN 68162 93638");
// c1 = {S 90º 0' 0" E 180º 0' 0"}
Coordinate c2 = Coordinate.Parse("23JMN6816293638");
// c2 = {S 24º 28' 14.021" W 45º 18' 51.033"}
```

**Expected behavior**
I'd expect `c1` to be the same value as `c2`.

**Environment (please complete the following information):**
- .NET Framework 4.7.2
- CoordinateSharp Version 2.8.1.1, Runtime v4.0.30319.
- Found in randomized Microsoft.VisualStudio.TestTools.UnitTesting-tests for a Windows Forms app.
- No multithreading in the unit tests

**Additional context**

After cloning I found that SpecialSplit (FormatFinder.Coordinate.cs, Ln 245+246) replaces the "MN"/"ME" (I assume for  UTM 23J 468162**mE** 7293637**mN**).

**Comment by Tronald** [comment]
Thanks for the report. Removing from SpecialSplit into specific parsers. Parsers were originally case sensitive, and when the switch was madeto case insensitive the identifiers should have been moved out of special split. Will fix thank you.

**Comment by B4rrCH** [comment]
Great.

I hope I'm not too bothersome with my edge cases. :-)

**Comment by Tronald** [comment]
Absolutely not! This helps us make a more reliable product, so all findings are greatly appreciated.

**Comment by Tronald** [comment]
Migrated UTM/MGRS to a more regex based solution that does not rely on the `SpecialSplit`. Much of the parser logic is outdated and can be streamlined as parser improvements are made. It should be noted that UTM and MGRS centimeter is not parseable at this time. 

Will be pushed with 2.9.1.1

**Comment by Tronald** [comment]
We had a version control issue, but an Alpha version of the package needed to be released anyway. 

It should be good for the most part, but the main concern is that we have not completed tests for new features and bug corrections. Nor have we complete documentation. All existing tests have passed though and manual testing has occurred. If you wish to test it out here it is. 

It should fix both reported bugs.

https://www.nuget.org/packages/CoordinateSharp/2.9.1.2-alpha

**Comment by B4rrCH** [comment]
I just tried the alpha, but it seems now that UTM parsing can fail now.

**To Reproduce**
```csharp 
Coordinate c1 = new Coordinate(5.209435694361094, -39.91750739404548);
// c1 = {N 5º 12' 33.968" W 39º 55' 3.027"}
string utm = c1.UTM.ToString();
Coordinate c2 = Coordinate.Parse(utm);
// Throws FormatException: Input Coordinate "24N 398319mE 575889mN" was not in a correct format.
```

With 2.8.1.1 `utm` is the same string, `c2` succeeds to be constructed and has identical values (at least the debugger representation) to `c1`.

**Comment by Tronald** [comment]
Yep, good catch again (that's why it's an alpha). This is a string length issue in the original regex part of the parser to determine if precision was unknown. I'll have to look into that one.

**Comment by Tronald** [comment]
We're adding all these test cases, so thank you for your contribution. UTM get's funky with string precision and the new tests weren't built yet to catch that against the Regex.

Check this package out (it's still indexing and may not show for a bit): https://www.nuget.org/packages/CoordinateSharp/2.9.1.3-alpha

Following examples parse safely:

```C#
            Console.WriteLine(Coordinate.Parse("23J MN 68162 93638").MGRS);
            Console.WriteLine(Coordinate.Parse("23JMN 68162 93638").MGRS);
            Console.WriteLine(Coordinate.Parse("23 JMN 6816293638").MGRS);
            Console.WriteLine(Coordinate.Parse("23 J MN 68162 93638").MGRS);
            Console.WriteLine(Coordinate.Parse("23JMN 68162mE 93638mN").MGRS);

            Console.WriteLine(Coordinate.Parse("   Z JG 53101 42860").MGRS);
            Console.WriteLine(Coordinate.Parse("ZJG 5310142860").MGRS);
            Console.WriteLine(Coordinate.Parse("0Z JG 53101ME 42860mn").MGRS);
            Console.WriteLine(Coordinate.Parse("ZJG53101ME 42860Mn").MGRS);
            Console.WriteLine(Coordinate.Parse("2 ZJG53101ME 42860Mn").MGRS);

            Console.WriteLine(Coordinate.Parse("Z 2653101mE 1942860mN").UTM);
            Console.WriteLine(Coordinate.Parse("Z26531011942860").UTM);
            Console.WriteLine(Coordinate.Parse(" Z 2653101mE 1942860mN").UTM);
            Console.WriteLine(Coordinate.Parse("Z26531011942860").UTM);


            Console.WriteLine(Coordinate.Parse("24N 398319mE 575889mN").UTM);           
            Console.WriteLine(Coordinate.Parse("24N 398319 575889").UTM);
            Console.WriteLine(Coordinate.Parse("43 V 500000mE 6874180mN").UTM);
            Console.WriteLine(Coordinate.Parse("43V5000006874180").UTM);
            Console.WriteLine(Coordinate.Parse("43V 500000 6874180").UTM);
```

I eventually want to build a more efficient parser, but they are so finnicky sometimes (as you have just seen). I appreciate your patients and feedback!

**Comment by B4rrCH** [comment]
Great, 2.9.1.3-alpha does seem to fix this as well as 170. 👍

**Comment by Tronald** [comment]
Fixed with v2.9.1.3 published on 11/29/2020.
